xlab/data/importer/iex/importer.py

```diff
@@ -56,3 +55,31 @@ def _make_data_base(self, symbol: str, data_date: datetime.datetime,
         data_entry.timestamp.FromDatetime(data_date)
         data_entry.updated_at.FromDatetime(now)
         return data_entry
+
+    def _get_date_range(self, start_date: datetime.datetime, end_date: datetime.datetime):
+        '''
+        calculate date range by counting no. of days between start and end days
+        '''
+
+        date_range = {
+            (0, 6): '5d',
+            (6, 28): '1m',
+            (28, 84): '3m',
+            (84, 168): '6m',
+            (168, 365): '1y',
+            (365, 730): '2y',
+            (730, 1826): '5y',
+            (1826, 5478): 'max',
+        }
+
+        timedelta = (end_date - start_date).days
+        timedelta_to_today = (datetime.datetime.now() - start_date).days
+
+        if timedelta < 0:
+            raise ValueError('Start date cannot be smaller than end date!')
+        elif timedelta_to_today >= 5478:
+            raise ValueError('Start date must be within past 15 years!')
+        else:
```

Comment:
This `else` statement is unnecessary.

Reply:
which one? the one with timedelta_to_today>=5478?

Reply:
Sure. Both the `elif` and the `else`. The raise will stop the execution of the current frame / scope, so there is no need to nest the normal execution in a nested block.
```
if timedelta < 0:
    raise ValueError('Start date cannot be smaller than end date!')
if timedelta_to_today >= 5478:
    raise ValueError('Start date must be within past 15 years!')

for key, value in date_range.items():
    if key[0] <= timedelta < key[1]:
        return value

```